sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -59,11 +63,15 @@ impl BlobClient {
             .per_try_policies
             .push(Arc::new(oauth_token_policy) as Arc<dyn Policy>);
 
-        let client = GeneratedBlobClient::new(endpoint, credential, Some(options))?;
+        let client = GeneratedBlobClient::new(
+            endpoint,
+            credential.clone(),
+            container_name.clone(),
+            blob_name.clone(),
+            Some(options.clone()),
```

Comment:
You were not cloning the options before, now you are? Did something change or can we remove the clone?

Reply:
Unnecessary clone, removed